### PR TITLE
Segment-rounting : Add documentation about segment-routing mpls on VyOS

### PR DESCRIPTION
**Pull request by fett0**
add documentation how to enable SR-MPLS in both IGP, ISIS-OSPF on VyOS

### CODE REVIEW

File: docs/configuration/protocols/segment-routing.rst [hunk 1]
```diff
@@ -0,0 +1,564 @@
+.. _segment-routing:
+
+###############
+Segment Routing
+###############
+
+Segment Routing (SR) is a network architecture that is similar to 
+source-routing.In this architecture, the ingress router adds a list of segments,
```
Comment (rebortg) [inline docs/configuration/protocols/segment-routing.rst:8]: missing space.

File: docs/configuration/protocols/segment-routing.rst [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+
+  No support for binding SID
+
+  No support for SRLB
+
+  Only one SRGB and default SPF Algorithm is supported
+
+
+
+.. cfgcmd:: set protocols isis segment-routing global-block <high-label-value
```
Comment (rebortg) [inline docs/configuration/protocols/segment-routing.rst:53]: this is not the correct syntax, same for local-block
this should be right.:

```
.. cfgcmd:: set protocols isis segment-routing global-block high-label-value <label-value>
.. cfgcmd:: set protocols isis segment-routing global-block low-label-value <label-value>
```

File: docs/configuration/protocols/segment-routing.rst [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+  <0-65535>
+   
+  A segment ID that contains an IP address prefix calculated by an IGP in the
+  service provider core network. Prefix SIDs are globally unique, this value
+  indentify it 
+
+.. cfgcmd:: set protocols isis segment-routing prefix <address> index
+   <no-php-flag | explicit-null| n-flag-clear>
+
+   prefix. The ‘no-php-flag’ means NO Penultimate Hop Popping that allows SR
```
Comment (rebortg) [inline docs/configuration/protocols/segment-routing.rst:83]: i think the `prefix.` belong the above block?

File: docs/configuration/protocols/segment-routing.rst [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+
+OSPF SR  Configuration
+----------------------
+
+Segment routing (SR) is used by the IGP protocols to interconnect network
+devices, below configuration shows how to enable SR on OSPF:
+
+.. cfgcmd:: set protocols ospf parameters opaque-lsa
+
+  Enable the Opaque-LSA capability (rfc2370),necessary to transport label
```
Comment (rebortg) [inline docs/configuration/protocols/segment-routing.rst:107]: please add space after the `,`

File: docs/configuration/protocols/segment-routing.rst [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+         T - Table, v - VNC, V - VNC-Direct, A - Babel, F - PBR,
+         f - OpenFabric,
+         > - selected route, * - FIB route, q - queued, r - rejected, b - backup
+         t - trapped, o - offload failure
+
+  O>* 10.1.1.1/32 [110/1] via 192.168.0.1, eth0, label IPv4 Explicit Null, weight 1, 00:03:36
+  O   10.1.1.2/32 [110/0] is directly connected, lo, weight 1, 00:03:51
+  O   192.168.0.0/24 [110/1] is directly connected, eth0, weight 1, 00:03:51
+
+**************************
```
Comment (rebortg) [inline docs/configuration/protocols/segment-routing.rst:322]: could you put the example in the [example](https://github.com/vyos/vyos-documentation/tree/master/docs/configexamples) section?

**Review by rebortg — CHANGES_REQUESTED** [review]

File: docs/configuration/protocols/segment-routing.rst (hunk 1 above)
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+which are running on most ISPs.
+
+
+.. note:: Segment routing defines a control plane network architecture and
+  can be applied to an existing MPLS based dataplane. In the MPLS networks,
+  segments are encoded as MPLS labels and are imposed at the ingress router.
+  MPLS labels are exchanged and populated by IGPs like IS-IS.Segment Routing
+  as per RFC8667 for MPLS dataplane. It supports IPv4, IPv6 and ECMP and has
+  been tested against Cisco & Juniper routers.however,this deployment is still
+  EXPERIMENTAL for FRR .
```
[the remark this answers is not in the document]
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst (hunk 2 above)
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst (hunk 3 above)
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst (hunk 4 above)
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+Segment routing (SR) is used by the IGP protocols to interconnect network
+devices, below configuration shows how to enable SR on OSPF:
+
+.. cfgcmd:: set protocols ospf parameters opaque-lsa
+
+  Enable the Opaque-LSA capability (rfc2370),necessary to transport label 
+  on IGP
+
+
+.. cfgcmd:: set protocols ospf segment-routing global-block <high-label-value
```
[the remark this answers is not in the document]
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+  <0-65535>
+   
+  A segment ID that contains an IP address prefix calculated by an IGP in the
+  service provider core network. Prefix SIDs are globally unique, this value
+  indentify it 
+
+.. cfgcmd:: set protocols ospf segment-routing prefix <address> index
+   <no-php-flag | explicit-null| n-flag-clear>
+
+   prefix. The ‘no-php-flag’ means NO Penultimate Hop Popping that allows SR
```
[the remark this answers is not in the document]
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

File: docs/configuration/protocols/segment-routing.rst (hunk 5 above)
Comment (fett0) [reply docs/configuration/protocols/segment-routing.rst]: done

**Pull request merged**